src/Build/Evaluation/Expander.cs

```diff
@@ -4456,21 +4456,25 @@ private static void ConstructFunction(IElementLocation elementLocation, string e
                 string[] functionArguments;
 
                 // The name of the function that will be invoked
-                string functionName;
+                ReadOnlySpan<char> functionName;
 
                 // What's left of the expression once the function has been constructed
-                string remainder = String.Empty;
+                ReadOnlySpan<char> remainder = ReadOnlySpan<char>.Empty;
 
                 // The binding flags that we will use for this function's execution
                 BindingFlags defaultBindingFlags = BindingFlags.IgnoreCase | BindingFlags.Public;
 
+                ReadOnlySpan<char> expressionFunctionAsSpan = expressionFunction.AsSpan();
+                
+                ReadOnlySpan<char> expressionSubstringAsSpan = argumentStartIndex > -1 ? expressionFunctionAsSpan.Slice(methodStartIndex, argumentStartIndex - methodStartIndex) : ReadOnlySpan<char>.Empty;
+
                 // There are arguments that need to be passed to the function
-                if (argumentStartIndex > -1 && !expressionFunction.Substring(methodStartIndex, argumentStartIndex - methodStartIndex).Contains("."))
+                if (argumentStartIndex > -1 && !expressionSubstringAsSpan.Contains(".".AsSpan(), StringComparison.OrdinalIgnoreCase))
```

Comment:
Does this overload exist? Feels cleaner if it does

```suggestion
                if (argumentStartIndex > -1 && !expressionSubstringAsSpan.Contains('.', StringComparison.OrdinalIgnoreCase))
```

Reply:
Unfortunately there are no other overloads for Contains(), it requires another ReadOnlySpan<char>

Reply:
@ahsonkhan  was this considered? I do not see use of StringComparison in SpanHelpers.Char.cs.

Reply:
We already have the following APIs around Contains/StringComparison:

Added in 2.1 as part of adding string-like APIs: https://github.com/dotnet/corefx/issues/21395
```C#
public static bool Contains(this ReadOnlySpan<char> span, ReadOnlySpan<char> value, StringComparison comparison);
```
Added in 3.0 as an optimization over IndexOf: https://github.com/dotnet/corefx/issues/27526 
```C#
bool Contains<T>(this Span<T> span, T value);
bool Contains<T>(this ReadOnlySpan<T> span, T value);
```

I believe the request is for an API that takes a single char instead of a `ROS<char>` like this (analogous to the string API https://apisof.net/catalog/System.String.Contains(Char,StringComparison)):
```C#
public static bool Contains(this ReadOnlySpan<char> span, char value, StringComparison comparison);
```

> I do not see use of StringComparison in SpanHelpers.Char.cs.

It's in MemoryExtensions.Fast.cs:
https://github.com/dotnet/coreclr/blob/c8e9c122aa5ea827e734fdffd889afa6318619da/src/System.Private.CoreLib/shared/System/MemoryExtensions.Fast.cs#L25-L28

> was this considered?

I don't remember exactly and it might have been an oversight, since most of the Contains APIs were added based on demonstrable need (also based on IndexOf APIs which also doesn't have an overload for a single char). I don't think we added span versions of all string-like APIs. This is probably one that we missed. The workaround is fairly straightforward though.
Either:
1) `".".AsSpan()` which gives a length-1 `ROSpan<char>`, as done here.
2) stackalloc a single char array that contains the lookup char and wrap that in a `ROSpan<char>`

@stephentoub, @GrabYourPitchforks - do you remember if this was discussed or if there is any reason we shouldn't consider adding such an overload? If we were to add it, we should probably add {Last}IndexOf overloads as well.

Reply:
Why does it need to be `".".AsSpan()` rather than just `"."`... the implicit cast isn't sufficient?

> any reason we shouldn't consider adding such an overload?

Primarily just the sheer number of overloads we're now talking about, every one of which has additional conceptual overhead, and we're talking about the difference between `"."` and `'.'`.